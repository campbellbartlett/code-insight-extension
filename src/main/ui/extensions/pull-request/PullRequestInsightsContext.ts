export type CodeInsightReport = {
    name: string;
    status: string;
};

export type PullRequestInsightsContext = {
    codeInsightReports: [CodeInsightReport];
    commitHash: string;
    repositorySlug: string;
    projectKey: string;
    userAdmin: boolean;
    adminOverride: boolean;
};
