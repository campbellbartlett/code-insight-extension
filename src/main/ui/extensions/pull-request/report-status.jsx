import React from 'react';
import PropTypes from 'prop-types';
import { Status } from '@atlaskit/status/element';

const StatusInParagraph = ({ text, color }) => (
    <div style={{ width: '4em' }}>
        <Status text={text} color={color} />
    </div>
);

const WaitingStatus = () => (
    <div>
        <StatusInParagraph text="Waiting" color="yellow" />
    </div>
);

const FailedStatus = () => (
    <div>
        <StatusInParagraph text="Failed" color="red" />
    </div>
);

const PassedStatus = () => (
    <div>
        <StatusInParagraph text="Passed" color="green" />
    </div>
);

const SingleReportStyle = { display: 'flex', width: '10rem', justifyContent: 'space-between' };

const SingleReportStatus = ({ report }) => {
    let icon;
    switch (report.status) {
        case 'WAITING':
            icon = <WaitingStatus />;
            break;
        case 'FAIL':
            icon = <FailedStatus />;
            break;
        case 'PASS':
            icon = <PassedStatus />;
            break;
        default:
            return <div />;
    }
    return (
        <div style={SingleReportStyle}>
            {report.name} {icon}
        </div>
    );
};

export const ReportStatus = ({ codeInsightReports }) => {
    return (
        <div style={{ marginTop: '1em', marginBottom: '1em' }}>
            <ul style={{ listStyleType: 'none' }}>
                {codeInsightReports.map(report => (
                    <li key={report.name}>
                        <SingleReportStatus report={report} />
                    </li>
                ))}
            </ul>
        </div>
    );
};

ReportStatus.propTypes = {
    codeInsightReports: PropTypes.arrayOf(
        PropTypes.oneOfType([PropTypes.string, PropTypes.string])
    ),
};

ReportStatus.defaultProps = {
    codeInsightReports: [],
};

StatusInParagraph.propTypes = {
    text: PropTypes.string,
    color: PropTypes.string,
};

StatusInParagraph.defaultProps = {
    text: '',
    color: 'neutral',
};

SingleReportStatus.propTypes = {
    report: PropTypes.objectOf([PropTypes.string, PropTypes.string]),
};

SingleReportStatus.defaultProps = {
    report: { name: '', status: '' },
};
