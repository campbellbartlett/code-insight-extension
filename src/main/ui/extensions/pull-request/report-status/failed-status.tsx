import React from 'react';
import StatusInParagraph from './status-in-paragraph';

const FailedStatus = () => <StatusInParagraph text="Failed" color="red" />;

export default FailedStatus;
