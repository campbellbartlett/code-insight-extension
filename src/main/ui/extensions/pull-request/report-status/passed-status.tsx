import React from 'react';
import StatusInParagraph from './status-in-paragraph';

const PassedStatus = () => <StatusInParagraph text="Passed" color="green" />;

export default PassedStatus;
