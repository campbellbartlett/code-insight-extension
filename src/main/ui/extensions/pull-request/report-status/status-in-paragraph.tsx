import React from 'react';
import { Status } from '@atlaskit/status/element';

type StatusInParagraphProps = {
    text: string;
    color: string;
};

const StatusInParagraph: React.FC<StatusInParagraphProps> = (props: StatusInParagraphProps) => {
    const { text, color } = props;
    return (
        <div style={{ width: '4em' }}>
            <Status text={text} color={color} />
        </div>
    );
};

export default StatusInParagraph;
