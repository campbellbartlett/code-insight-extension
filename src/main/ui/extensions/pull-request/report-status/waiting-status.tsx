import React from 'react';
import StatusInParagraph from './status-in-paragraph';

const WaitingStatus = () => <StatusInParagraph text="Waiting" color="yellow" />;

export default WaitingStatus;
