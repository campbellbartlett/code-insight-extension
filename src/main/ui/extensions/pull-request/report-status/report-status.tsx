import React from 'react';
import { CodeInsightReport } from '../PullRequestInsightsContext';
import SingleReportStatus from './single-report-status';

type ReportStatusProps = {
    codeInsightReports: Array<CodeInsightReport>;
};

export const ReportStatus: React.FC<ReportStatusProps> = (props: ReportStatusProps) => {
    const { codeInsightReports } = props;
    return (
        <div id="report-status" style={{ marginTop: '1em', marginBottom: '1em' }}>
            <ul style={{ listStyleType: 'none' }}>
                {codeInsightReports.map(report => {
                    // Use report name as id (will be unque due to BitBucketServer contraint but remove whitespace)
                    const id = report.name.replace(' ', '');
                    return (
                        <li id={id} key={id}>
                            <SingleReportStatus report={report} />
                        </li>
                    );
                })}
            </ul>
        </div>
    );
};
