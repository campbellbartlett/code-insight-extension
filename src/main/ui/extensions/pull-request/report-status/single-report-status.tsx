import React from 'react';
import { CodeInsightReport } from '../PullRequestInsightsContext';
import WaitingStatus from './waiting-status';
import FailedStatus from './failed-status';
import PassedStatus from './passed-status';

const SingleReportStyle = { display: 'flex', width: '10rem', justifyContent: 'space-between' };

type SingleReportProps = {
    report: CodeInsightReport;
};

const SingleReportStatus: React.FC<SingleReportProps> = (props: SingleReportProps) => {
    const { report } = props;
    let icon;
    switch (report.status) {
        case 'WAITING':
            icon = <WaitingStatus />;
            break;
        case 'FAIL':
            icon = <FailedStatus />;
            break;
        case 'PASS':
            icon = <PassedStatus />;
            break;
        default:
            return <div />;
    }
    return (
        <div style={SingleReportStyle}>
            <span id="report-name">{report.name}</span> {icon}
        </div>
    );
};

export default SingleReportStatus;
