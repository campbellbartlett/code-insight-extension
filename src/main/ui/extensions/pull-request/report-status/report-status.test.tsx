import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import { ReportStatus } from './report-status';
import { CodeInsightReport } from '../PullRequestInsightsContext';

const insightReports: Array<CodeInsightReport> = [
    {
        status: 'PASS',
        name: 'Report 1',
    },
    {
        status: 'FAIL',
        name: 'Report 2',
    },
    {
        status: 'WAITING',
        name: 'Report 3',
    },
];

describe('report status component shows correct data', () => {
    it('should render correctly', () => {
        const reportStatus = shallow(<ReportStatus codeInsightReports={insightReports} />);

        expect(shallowToJson(reportStatus)).toMatchSnapshot();
    });

    it('should render the correct number of report statuses', () => {
        const reportStatus = shallow(<ReportStatus codeInsightReports={insightReports} />);

        const reportStatusItems = reportStatus.find('li');
        expect(reportStatusItems.length).toEqual(3);

        expect(reportStatus.find('#Report1').length).toEqual(1);
        expect(reportStatus.find('#Report2').length).toEqual(1);
        expect(reportStatus.find('#Report3').length).toEqual(1);
    });
});
