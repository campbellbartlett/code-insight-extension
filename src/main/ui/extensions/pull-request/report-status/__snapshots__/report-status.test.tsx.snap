// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report status component shows correct data should render correctly 1`] = `
<div
  id="report-status"
  style={
    Object {
      "marginBottom": "1em",
      "marginTop": "1em",
    }
  }
>
  <ul
    style={
      Object {
        "listStyleType": "none",
      }
    }
  >
    <li
      id="Report1"
      key="Report1"
    >
      <SingleReportStatus
        report={
          Object {
            "name": "Report 1",
            "status": "PASS",
          }
        }
      />
    </li>
    <li
      id="Report2"
      key="Report2"
    >
      <SingleReportStatus
        report={
          Object {
            "name": "Report 2",
            "status": "FAIL",
          }
        }
      />
    </li>
    <li
      id="Report3"
      key="Report3"
    >
      <SingleReportStatus
        report={
          Object {
            "name": "Report 3",
            "status": "WAITING",
          }
        }
      />
    </li>
  </ul>
</div>
`;
